### PR TITLE
Add more keyword arguments to PercivalSolver

### PR DESCRIPTION
**Pull request by tmigot**

### CODE REVIEW

File: src/method.jl [hunk 1]
```diff
@@ -141,7 +141,11 @@ mutable struct PercivalSolver{V} <: AbstractOptimizationSolver
   sub_solver::TronSolver
 end
 
-function PercivalSolver(nlp::AbstractNLPModel{T, V}; subproblem_modifier = identity) where {T, V}
+function PercivalSolver(
+  nlp::AbstractNLPModel{T, V};
+  subproblem_modifier = identity,
+  kwargs...,
```
Comment (abelsiqueira) [inline src/method.jl:147]:
```suggestion
  subsolver_kwargs = Dict(:max_cgiter => nlp.meta.nvar),
```
Like the other calls.

File: src/method.jl [hunk 2]
```diff
@@ -150,14 +154,26 @@ function PercivalSolver(nlp::AbstractNLPModel{T, V}; subproblem_modifier = ident
   Jtv = V(undef, nvar)
 
   sub_pb = AugLagModel(nlp, V(undef, ncon), T(0), x, T(0), V(undef, ncon))
-  sub_solver = TronSolver(subproblem_modifier(sub_pb))
+  sub_solver = TronSolver(subproblem_modifier(sub_pb); kwargs...)
```
Comment (abelsiqueira) [inline src/method.jl:157]:
```suggestion
  sub_solver = TronSolver(subproblem_modifier(sub_pb); subsolver_kwargs...)
```

File: src/method.jl [hunk 3]
```diff
@@ -168,8 +184,14 @@ end
...
-  solver = PercivalSolver(nlp, subproblem_modifier = subproblem_modifier)
-  SolverCore.solve!(solver, nlp; subproblem_modifier = subproblem_modifier, kwargs...)
+  subsolver_keys = intersect(keys(subsolver_kwargs), percival_keys)
+  solver_kwargs = Dict(k => subsolver_kwargs[k] for k in subsolver_keys)
+  solver = PercivalSolver(nlp; subproblem_modifier = subproblem_modifier, solver_kwargs...)
+  sub_kwargs = copy(subsolver_kwargs)
+  for k in subsolver_keys
+    pop!(sub_kwargs, k)
+  end
+  SolverCore.solve!(solver, nlp; subproblem_modifier = subproblem_modifier, subsolver_kwargs = sub_kwargs, kwargs...)
```
Comment (abelsiqueira) [inline src/method.jl:194]: I think you can leave almost everything as it was, except pass the `subsolver_kwargs` to `PercivalSolver` (`solve!` won't need it, right?).
That also mean removing the `percival_keys` definition.

**Review by abelsiqueira — CHANGES_REQUESTED** [review]
Thanks for the PR. Can you try using only `subsolver_kwargs`?

File: src/method.jl (hunk 3 above)
Comment (tmigot) [reply src/method.jl:194]: Actually, some keywords go to the `PercivalSolver` and some in the `solve!`.
For instance, `:max_cgiter` goes to the `solve!` while `max_radius` goes to `PercivalSolver` (and then to `TronSolver`).
So, passing `max_cgiter` to `PercivalSolver` would return an error.

Comment (abelsiqueira) [reply src/method.jl:194]: Ah, ok. I misread the `tron` changes. I would have preferred to have something similar in `tron`, but I guess it should be changed when we move to AbstractTrustRegion.
Can you change the name of `percival_keys` to `subsolver_keys` or `trustregion_keys`?

Comment (tmigot) [reply src/method.jl]: Agreed, for updating the changes when moving to `AbstractTrustRegion`.

Comment (tmigot) [reply src/method.jl]: But, there will always be a similar problem as some parameters are for defining the `TronSolver` structure, while others will be for the `solve!` in `tron`.

Comment (tmigot) [reply src/method.jl]: I renamed `percival_keys` as `trustregion_keys`

**Review by abelsiqueira — APPROVED** [review]
We should add to our todo list to discuss how to handle that. Line searches should have a similar approach. For now, I approved this PR, thanks.

**Pull request merged**